Framework/Algorithms/test/MaxEnt/MaxentTransformMultiFourierTest.h

```diff
@@ -0,0 +1,232 @@
+    MaxentSpaceComplex_sptr dataSpaceMF =
+        boost::make_shared<MaxentSpaceComplex>();
+    MaxentSpace_sptr imageSpaceMF = boost::make_shared<MaxentSpaceComplex>();
+    MaxentTransformMultiFourier transformMF(dataSpaceMF, imageSpaceMF, 3);
+    MaxentSpace_sptr dataSpaceF = boost::make_shared<MaxentSpaceComplex>();
+    MaxentSpace_sptr imageSpaceF = boost::make_shared<MaxentSpaceComplex>();
+    MaxentTransformFourier transformF(dataSpaceF, imageSpaceF);
+
+    std::vector<double> complexImage = {4.0, 3.0, 2.0, 1.0, 0.0, 0.0, 0.0, 0.0};
+    std::vector<double> linearAdjustments = {
+        0.0, 1.0, 0.0, 2.0, 0.0, 3.0, 0.0, 4.0, 0.0, 0.0, 0.0, 0.0,
+        0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 2.0, 0.0, 3.0, 0.0, 4.0};
+    std::vector<double> constAdjustments = {
+        0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,  0.0, 1.0, 0.0, 2.0,
+        0.0, 3.0, 0.0, 4.0, 0.0, 1.0, 0.0, -1.0, 0.0, 1.0, 0.0, -1.0};
+
+    // Set Adjustments
+    transformMF.setAdjustments(linearAdjustments, constAdjustments);
+
+    // Perform the transformation
+    auto result = transformMF.imageToData(complexImage);
+    // Perform Fourier transformation for comparison
+    auto resultF = transformF.imageToData(complexImage);
+
+    // Check that result is a triple repetition of fourier
+    // but with the adjustments applied
+    TS_ASSERT_EQUALS(result.size(), 3 * resultF.size());
+    TS_ASSERT_EQUALS(result.size(), linearAdjustments.size());
+    TS_ASSERT_EQUALS(result.size(), constAdjustments.size());
+
+    if (result.size() == 3 * resultF.size()) {
+      size_t n = resultF.size();
+      for (size_t i = 0; i < n; i++) {
+        if (i % 2 == 0) { // Real part
+          TS_ASSERT_DELTA(result[i],
+                          -linearAdjustments[i + 1] * resultF[i + 1] +
+                              constAdjustments[i],
+                          1e-4);
+          TS_ASSERT_DELTA(result[i + n],
+                          -linearAdjustments[i + n + 1] * resultF[i + 1] +
+                              constAdjustments[i + n],
+                          1e-4);
+          TS_ASSERT_DELTA(result[i + 2 * n],
+                          -linearAdjustments[i + 2 * n + 1] * resultF[i + 1] +
+                              constAdjustments[i + 2 * n],
+                          1e-4);
+        } else { // Imaginary part
+          TS_ASSERT_DELTA(result[i], linearAdjustments[i] * resultF[i - 1] +
+                                         constAdjustments[i],
+                          1e-4);
+          TS_ASSERT_DELTA(result[i + n],
+                          linearAdjustments[i + n] * resultF[i - 1] +
+                              constAdjustments[i + n],
+                          1e-4);
+          TS_ASSERT_DELTA(result[i + 2 * n],
+                          linearAdjustments[i + 2 * n] * resultF[i - 1] +
+                              constAdjustments[i + 2 * n],
+                          1e-4);
+        }
+      }
+    }
+  }
+};
```

Comment:
unless I have missed something, it looks like the values for transform is never tested as it is always compared against another transform. So if the alg changes it  could still pass even if it gives a bad answer.

Reply:
Which alg?

Reply:
your ```dataToImage``` method is the one that stood out. There might be others too

Reply:
This needs explaining in more detail.

Reply:
If either the `dataToImage` or `imageToData` methods of `MaxentTransformMultiFourier` were to give bad answer, it would show up against a comparison with the same method of `MaxentTransformFourier`. If you think I'm wrong here, please give a senario where a bad result would not cause the test to fail.

Reply:
could you please give me a link to the line of code you think check the transform values?

Reply:
I only compare against the values given by the plain fourier transform.